strawberryfields/configuration.py

```diff
@@ -0,0 +1,189 @@
+
+    [api]
+    # Options for the Strawberry Fields Cloud API
+    authentatication_token = "071cdcce-9241-4965-93af-4a4dbc739135"
+    hostname = "localhost"
+    use_ssl = true
+
+Summary of options
+------------------
+
+todo
+
+Summary of methods
+------------------
+
+.. currentmodule:: strawberryfields.configuration.Configuration
+
+.. autosummary::
+    path
+    load
+    save
+
+Code details
+~~~~~~~~~~~~
+
+.. currentmodule:: strawberryfields.configuration
+
+"""
+import os
+import logging as log
+
+import toml
+from appdirs import user_config_dir
+
+log.getLogger()
+
+
+DEFAULT_CONFIG = {"api": {"authentatication_token": "", "hostname": "localhost", "use_ssl": True}}
+
+
+class ConfigurationError(Exception):
+    """Exception used for configuration errors"""
+
+
+class Configuration:
+    """Configuration class.
+
+    This class is responsible for loading, saving, and storing StrawberryFields
+    and plugin/device configurations.
+
+    Args:
+        name (str): filename of the configuration file.
+        This should be a valid TOML file. You may also pass an absolute
+        or a relative file path to the configuration file.
+    """
+
+    def __str__(self):
+        return "{}".format(self._config)
+
+    def __repr__(self):
+        return "Strawberry Fields Configuration <{}>".format(self._filepath)
+
+    def __init__(self, name="config.toml"):
+        # Look for an existing configuration file
+        self._config = DEFAULT_CONFIG
+        self._config_file = {}
+        self._filepath = None
+        self._name = name
+        self._user_config_dir = user_config_dir("strawberryfields", "Xanadu")
+        self._env_config_dir = os.environ.get("SF_CONF", "")
+
+        # search the current directory, the directory under environment
+        # variable SF_CONF, and default user config directory, in that order.
+        directories = [os.curdir, self._env_config_dir, self._user_config_dir, ""]
+        for idx, directory in enumerate(directories):
+            try:
+                self._filepath = os.path.join(directory, self._name)
+                self.load(self._filepath)
+                break
+            except FileNotFoundError:
+                if idx == len(directories) - 1:
+                    log.info("No Strawberry Fields configuration file found.")
+
+        self.update_config()
+
+    def update_config(self):
+        """Updates the configuration from either a loaded configuration
+        file, or from an environment variable.
+
+        The environment variable takes precedence."""
+        for section, section_config in self._config.items():
+            env_prefix = "SF_{}_".format(section.upper())
+
+            for key, _ in section_config.items():
+                # environment variables take precedence
+                env = env_prefix + key.upper()
+                if env in os.environ:
+                    self._config[section][key] = os.environ[env]
+                    continue
```

Comment:
I think it would be a little more readable if instead of continue here, we used `elif` to combine with the logic below.

Reply:
done